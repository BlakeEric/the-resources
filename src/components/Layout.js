import React from "react"
import { css, Global } from "@emotion/core"
import { useStaticQuery, Link, graphql } from "gatsby"
import { rhythm } from "../utils/typography"

export const PureLayout = ({ data, children }) => (
  <div css={wrapperStyles}>
    <Global styles={globalStyles} />
    <nav>
      <Link to={`/`}>
        <h3
          css={css`
            margin-bottom: ${rhythm(2)};
            display: inline-block;
            font-style: normal;
          `}
        >
          {data.site.siteMetadata.title}
        </h3>
      </Link>
      <Link
        to={`/about/`}
        css={css`
          float: right;
        `}
      >
        About
      </Link>
    </nav>
    <div className="page">
      {children}
    </div>
    <footer css={footerStyles}>
      <span>
        Â©Blake Lundquist {new Date().getFullYear()}
      </span>
      <ul>
        <li>
          <a href="mailto:blake@blakelundquist.dev">Recommend a resource</a></li>
        <li>
          <a target="_blank" rel="noopener noreferrer" href="https://github.com/BlakeEric/the-resources">
            View source code for this website
          </a>
        </li>
      </ul>
    </footer>
  </div>
)

export default function Layout({ children }) {
  const data = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `
  )
  return (
    <PureLayout data={data}>{ children }</PureLayout>
  )
}



const globalStyles = css`
  body {
    min-height: 100vh;
    padding: 0 ${rhythm(.25)};
  }
`


const wrapperStyles = css`
  margin: ${rhythm(.5)} auto;
  max-width: 900px;
  padding: ${rhythm(1)} ${rhythm(.5)};
  background: white;
  @media (min-width: 600px) {
    padding: ${rhythm(1.5)};
  }
  .page {
    min-height: 60vh;
  }
`


const footerStyles = css`
  margin-top: ${rhythm(2)};
  font-size: ${rhythm(.575)};
  @media (min-width: 600px) {
    display: flex;
    justify-content: space-between;
  }
  span {
    display: inline-block;
    margin-bottom: ${rhythm(.25)};
  }
  ul {
    list-style: none;
    margin: 0;
    @media (min-width: 600px) {
     text-align: right;
    }
    li {
      margin-bottom: ${rhythm(.25)};
      line-height: 1.25;
    }
  }
`
