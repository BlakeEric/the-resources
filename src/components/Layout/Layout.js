import React from "react"
import { css, Global } from "@emotion/core"
import { useStaticQuery, Link, graphql } from "gatsby"
import { Box, Flex, Container } from 'theme-ui'
import theme from '../../gatsby-plugin-theme-ui/index'


export const PureLayout = ({ data, children }) => (
  <>
    <Global
      styles={{
        a: theme.styles.a
      }}
    />

    <Container pt={3}>
      <nav>
        <Flex>
          <Box p={2} sx={{ flex: '1 1 auto' }}>
            <Link to="/" css={css`
              text-decoration: none;
            `}>
              <h3 css={css`
                margin: 0;
                line-height: 1.5;
              `}>
                {data.site.siteMetadata.title}
              </h3>
            </Link>
          </Box>
          <Box p={2} >
            <Link to="/about/">About</Link>
          </Box>
        </Flex>
      </nav>
    </Container>

    {/* Page content */}
    <Container p={2} sx={{minHeight: "80vh"}}>
      {children}
    </Container>
    {/* End page content */}

    <Container>
      <footer>
        <Flex pt={5} pb={5}>
          <Box p={2} sx={{ flex: '1 1 auto' }}>
            <span>
              Â© Blake Lundquist {new Date().getFullYear()}
            </span>
          </Box>
          <Box p={2}>
            <ul css={css`
              margin: 0;
              list-style: none;
              padding: 0;
            `}>
              <li>
                <a href="mailto:blake@blakelundquist.dev?subject=A new resource for theresources.dev!">Recommend a resource</a></li>
              <li>
                <a target="_blank" rel="noopener noreferrer" href="https://github.com/BlakeEric/the-resources">
                  View source code for this website
                </a>
              </li>
            </ul>
          </Box>
        </Flex>
      </footer>
    </Container>
  </>
)

export default function Layout({ children }) {
  const data = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `
  )
  return (
    <PureLayout data={data}>{ children }</PureLayout>
  )
}
