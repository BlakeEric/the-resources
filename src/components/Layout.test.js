import React from "react"
import renderer from "react-test-renderer"
import { PureLayout as Layout } from "./Layout"
import { LocationProvider, createHistory, createMemorySource } from "@reach/router";

describe("Layout", () => {

  const data = {
    site: {
      siteMetadata: {
        title: "Cool Site Title",
      },
    },
  }


  const testHistory = createHistory(createMemorySource("/"));


  it("renders correctly", () => {
    const tree = renderer.create(
      <LocationProvider history={testHistory}>
        <Layout path="/" data={data} />
      </LocationProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it("renders correctly with children", () => {
  //   const tree = renderer.create(
  //     <LocationProvider history={history}>
  //       <Layout data={data} path="/">
  //         <main>
  //           <h1>I'm a child!</h1>
  //         </main>
  //       </Layout>
  //     </LocationProvider>
  //   ).toJSON()
  //   expect(tree).toMatchSnapshot()
  // })

})
