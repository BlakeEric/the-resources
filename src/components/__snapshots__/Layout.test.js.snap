// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "vi9d7b",
      "next": undefined,
      "styles": "
  margin: 0.805rem auto;
  max-width: 900px;
  padding: 1.61rem 0.805rem;
  background: white;
  @media (min-width: 600px) {
    padding: 2.415rem;
  }
  .page {
    min-height: 60vh;
  }
",
      "toString": [Function],
    }
  }
>
  <nav>
    <a
      href="/"
    >
      <h3
        css={
          Object {
            "map": undefined,
            "name": "kgn0fp",
            "next": undefined,
            "styles": "
            margin-bottom: 3.22rem;
            display: inline-block;
            font-style: normal;
          ",
            "toString": [Function],
          }
        }
      >
        Cool Site Title
      </h3>
    </a>
    <a
      css={
        Object {
          "map": undefined,
          "name": "1huhmlv",
          "next": undefined,
          "styles": "
          float: right;
        ",
          "toString": [Function],
        }
      }
      href="/about/"
    >
      About
    </a>
  </nav>
  <div
    className="page"
  />
  <footer
    css={
      Object {
        "map": undefined,
        "name": "1vkm4mx",
        "next": undefined,
        "styles": "
  margin-top: 3.22rem;
  font-size: 0.92575rem;
  @media (min-width: 600px) {
    display: flex;
    justify-content: space-between;
  }
  span {
    display: inline-block;
    margin-bottom: 0.4025rem;
  }
  ul {
    list-style: none;
    margin: 0;
    @media (min-width: 600px) {
     text-align: right;
    }
    li {
      margin-bottom: 0.4025rem;
      line-height: 1.25;
    }
  }
",
        "toString": [Function],
      }
    }
  >
    <span>
      Â©Blake Lundquist 
      2020
    </span>
    <ul>
      <li>
        <a
          href="mailto:blake@blakelundquist.dev?subject=A new resource for theresources.dev!"
        >
          Recommend a resource
        </a>
      </li>
      <li>
        <a
          href="https://github.com/BlakeEric/the-resources"
          rel="noopener noreferrer"
          target="_blank"
        >
          View source code for this website
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
